docs/developer/best-practices/performance.asciidoc

```diff
@@ -0,0 +1,102 @@
+[[plugin-performance]]
+== Keep Kibana fast
+
+*tl;dr*: Load as much code lazily as possible. Everyone loves snappy
+applications with responsive UI and hates spinners. Users deserve the
+best user experiences regardless of whether they run Kibana locally or
+in the cloud, regardless of their hardware & environment.
+
+There are 2 main aspects of the perceived speed of an application: loading time
+and responsiveness to user actions. Kibana loads and bootstraps *all*
+the plugins whenever a user lands on any page. It means that
+every new application affects the overall _loading performance_, as plugin code is
+loaded _eagerly_ to initialize the plugin and provide plugin API to dependent
+plugins.
+
+However, it’s usually not necessary that the whole plugin code should be loaded
+and initialized at once. The plugin could keep on loading code covering API functionality
+on Kibana bootstrap but load UI related code lazily on-demand, when an
+application page or management section is mounted.
+Always prefer to
+require UI root components lazily when possible (such as in `mount`
```

Comment:
A plugin might require a module in runtime, but it doesn't control the `loading` mechanism -  Kibana does.
```js
const module = await import('./my_module'); <-- A plugin wants to get 'my_module' and knows it might take time to load the bundle, but it doesn't know how Kibana loads one.
```

Reply:
maybe we can use "import" instead of "require"